### ISSUE TITLE
Wrong display of `é`, `à` and `è` (and probably some other characters) as `�`

### ISSUE THREAD

**Issue by louisroyer** [00-Evan/shattered-pixel-dungeon#366]
Running [ShatteredPD-v0.8.0-BETA-3.2-Desktop.jar](https://github.com/00-Evan/shattered-pixel-dungeon/releases/download/xbeta/ShatteredPD-v0.8.0-BETA-3.2-Desktop.jar) on debian.
```
$ java -version
openjdk version "11.0.6" 2020-01-14
OpenJDK Runtime Environment (build 11.0.6+10-post-Debian-1deb10u1)
OpenJDK 64-Bit Server VM (build 11.0.6+10-post-Debian-1deb10u1, mixed mode, sharing)
```

![2020-01-30-204710_747x610_scrot](https://user-images.githubusercontent.com/55180044/73484513-bee9a400-43a1-11ea-962b-67bf3951222d.png)

**Comment by 00-Evan** [comment]
Odd, this works correctly on my end. I wonder if your system is using different codepoints for those glyphs or something. The game should give some console output whenever it fails to render a glyph, could you include that please?

**Comment by louisroyer** [comment]
Launching the game with `$ java -jar ShatteredPD-v0.8.0-BETA-3.2-Desktop.jar`.
I have no special console output when failing to render a glyph.
When starting a new game I get `[GAME] @@ Vous descendez � l'�tage 1 du donjon.` in console, but nothing more (and I am able to enter `é` in my console, it prints correctly).
Also, the `œ` is rendered as `?` (and not as `�`) in game’s first floor message (last line).

In case it could be usefull:
```
$ locale
LANG=fr_FR.UTF-8
LANGUAGE=
LC_CTYPE="fr_FR.UTF-8"
LC_NUMERIC="fr_FR.UTF-8"
LC_TIME="fr_FR.UTF-8"
LC_COLLATE="fr_FR.UTF-8"
LC_MONETARY="fr_FR.UTF-8"
LC_MESSAGES="fr_FR.UTF-8"
LC_PAPER="fr_FR.UTF-8"
LC_NAME="fr_FR.UTF-8"
LC_ADDRESS="fr_FR.UTF-8"
LC_TELEPHONE="fr_FR.UTF-8"
LC_MEASUREMENT="fr_FR.UTF-8"
LC_IDENTIFICATION="fr_FR.UTF-8"
LC_ALL=
```

**Comment by 00-Evan** [comment]
Okay, so then the game seems to be having some issues with reading the text files in, perhaps not correctly interpreting them as utf-8. The rendering of the text is working correctly, but for some reason that's the glyph that the game interprets from the file.

I've attached a debug JAR that adjusts how the text resource files are read, let me know if that works: 
[ShatteredPD v0.8.0-BETA-3.2-Desktop i18n debug.zip](https://github.com/00-Evan/shattered-pixel-dungeon/files/4141913/ShatteredPD.v0.8.0-BETA-3.2-Desktop.i18n.debug.zip)

**Comment by louisroyer** [comment]
Yes, it works both on game screen and in console output now. Thank you.

**Comment by 00-Evan** [comment]
Excellent. I'll include this fix in the next beta update.

**Comment by 00-Evan** [comment]
Should now be fixed as of beta-4.0
